include/Metabox/Posts/CustomTaxonomy.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Metabox to move posts based on category Class.
+ *
+ * @since 2.0.0
+ */
+class BM_Metabox_Posts_CustomTaxonomy extends BM_Metabox_Base {
+
+	protected function initialize() {
+		$this->meta_box_slug = 'bm-posts-by-custom-taxonomy';
+		$this->messages['box_label'] = __( 'Move Posts By Custom Taxonomy', 'bulk-move' );
+		$this->action = 'move_custom_taxonomy';
+	}
+
+	public function render() {
+		?>
+
+		<!-- Custom Taxonomy Start-->
+
+		<fieldset class="options">
+			<table class="optiontable">
+				<tr>
+					<td scope="row" colspan="2">
+						<?php _e( 'Select the post type to show its custom taxonomy.', 'bulk-move' ); ?>
+					</td>
+					<td scope="row">
+				</tr>
+
+				<tr>
+					<td scope="row" colspan="2">
+						<select name="smbm_mbct_post_type" id="smbm_mbct_post_type">
+							<option value="-1"><?php _e( 'Select Post type', 'bulk-move' ); ?></option>
+
+							<?php
+							$custom_post_types = get_post_types( array( 'public' => true ) );
+							?>
+
+							<?php foreach ( $custom_post_types as $post_type ) : ?>
+								<option value="<?php echo esc_attr( $post_type ); ?>"><?php echo esc_html( $post_type ); ?></option>
+							<?php endforeach; ?>
+						</select>
+					</td>
+				</tr>
+
+				<tr class="taxonomy-select-row">
+					<td scope="row" colspan="2">
+						<?php _e( 'Select taxonomy to show its terms.', 'bulk-move' ); ?>
+					</td>
+					<td scope="row">
+				</tr>
+
+				<tr class="taxonomy-select-row">
+					<td scope="row" colspan="2">
+						<select name="smbm_mbct_taxonomy" id="smbm_mbct_taxonomy">
+							<option value="select"><?php _e( 'Select Taxonomy', 'bulk-move' ); ?></option>
+						</select>
+					</td>
+				</tr>
+
+				<tr class="term-select-row">
+					<td scope="row" colspan="2">
+						<?php _e( 'Select terms to move its posts.', 'bulk-move' ); ?>
+					</td>
+					<td scope="row">
+				</tr>
+
+				<tr class="term-select-row">
+					<td scope="row" >
```

Comment:
Abstract this into a helper function that is added to the base class

Reply:
@sudar,
 
> Abstract this into a helper function that is added to the base class

The Select Term dropdown has only the `<select>` and a default `<option>` tag. Even if we abstract it out to the the base class, it will be dependent on the AJAX to load terms.

Do you think we should restructure the AJAX JS, or should we keep this Terms `<select>` dropdown as such?

Reply:
Let's just leave it like that.